pkg/connector/ble/iface/connection.go

```diff
@@ -0,0 +1,128 @@
+package iface
+
+import (
+	"context"
+	"github.com/teslamotors/vehicle-command/internal/log"
+	"github.com/teslamotors/vehicle-command/pkg/connector"
+	"sync"
+	"time"
+)
+
+const (
+	MaxBLEMessageSize = 1024
+
+	rxTimeout  = time.Second     // Timeout interval between receiving chunks of a message
+	maxLatency = 4 * time.Second // Max allowed error when syncing vehicle clock
+
+	VehicleServiceUUID = "00000211-b2d1-43f0-9b88-960cebf8b91e"
+	ToVehicleUUID      = "00000212-b2d1-43f0-9b88-960cebf8b91e"
+	FromVehicleUUID    = "00000213-b2d1-43f0-9b88-960cebf8b91e"
+)
+
+type Writer interface {
+	WriteCharacteristic(bytes []byte, length int) error
```

Comment:
This creates a bit of a hacky almost circular dependency. I think it would be cleaner to have the interface expose methods for for both reading (or exposing a channel for reading packets) and writing; the implmentation should be self-contained.

Reply:
I will try to re-think this. first step was to separate it out, but now I got more familiar with both versions, I will try to make it better